modules/websocket/websocket-destination.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2014 BalaBit IT Ltd, Budapest, Hungary
+ * Copyright (c) 2016 Yilin Li <liyilin1214@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "websocket-destination.h"
+#include "ws_api/client.h"
+#include "ws_api/server.h"
+#include <stdio.h>
+#include <string.h>
+
+
+gboolean
+websocket_dd_init(LogPipe *destination)
+{
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  GlobalConfig *cfg = log_pipe_get_config(destination);
+
+  if (!log_dest_driver_init_method(destination))
+    return FALSE;
+
+  log_template_options_init(&self->template_options, cfg);
+
+  if (self->enable_ssl)
+    self->client_use_ssl_flag = self->allow_self_signed ? 2 : 1;
+  else
+    self->client_use_ssl_flag = 0;
+
+  if (strcmp(self->mode, "client") && strcmp(self->mode, "server")) {
+    msg_error("Only \"server\" and \"client\" are supported for mode option!", evt_tag_str("mode", self->mode), NULL);
+    return FALSE;
+  }
```

Comment:
I think you shouldn't check the mode here.
Check it in `websocket_dd_set_mode` and use the `CHECK_ERROR` macro and the error will appear where it generated.